ClientTests/FeatureFlagsUserPrefsMigrationUtilityTests.swift

```diff
@@ -0,0 +1,200 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/
+
+import XCTest
+import Shared
+
+@testable import Client
+
+class FeatureFlagsUserPrefsMigrationUtilityTests: XCTestCase {
+
+    // MARK: - Properties
+    typealias legacyFlags = PrefsKeys.LegacyFeatureFlags
+    typealias newFlags = PrefsKeys.FeatureFlags
+
+    private let userPrefsSuffix = "UserPreferences"
+    private let keyDictionary = [
+        legacyFlags.ASPocketStories: newFlags.ASPocketStories,
+        legacyFlags.CustomWallpaper: newFlags.CustomWallpaper,
+        legacyFlags.HistoryHighlightsSection: newFlags.HistoryHighlightsSection,
+        legacyFlags.HistoryGroups: newFlags.HistoryGroups,
+        legacyFlags.InactiveTabs: newFlags.InactiveTabs,
+        legacyFlags.JumpBackInSection: newFlags.JumpBackInSection,
+        legacyFlags.PullToRefresh: newFlags.PullToRefresh,
+        legacyFlags.RecentlySavedSection: newFlags.RecentlySavedSection,
+        legacyFlags.SponsoredShortcuts: newFlags.SponsoredShortcuts,
+        legacyFlags.StartAtHome: newFlags.StartAtHome,
+        legacyFlags.TabTrayGroups: newFlags.TabTrayGroups,
+        legacyFlags.TopSiteSection: newFlags.TopSiteSection]
+
+    private var profile: MockProfile!
+
+    // MARK: - Set-up and tear down
+    override func setUp() {
+        super.setUp()
+
+        profile = MockProfile(databasePrefix: "migrationUtility_tests")
+        profile._reopen()
+    }
+
+    override func tearDown() {
+        super.tearDown()
+
+        profile._shutdown()
+        profile = nil
+    }
+
+    // MARK: - Tests
+
+    func testVerifyEmptyProfiles() {
+        verifyNilStateForKeys()
+    }
+
+    func testSetAllKeysToTrue() {
+        verifyNilStateForKeys()
+
+        keyDictionary.forEach { oldKey, _ in
+            let oldKey = oldKey + userPrefsSuffix
+            profile.prefs.setString(UserFeaturePreference.enabled.rawValue, forKey: oldKey)
+        }
+
+        keyDictionary.forEach { oldKey, newKey in
+            let oldKey = oldKey + userPrefsSuffix
+            let newKey = newKey
+            guard let string = profile.prefs.stringForKey(oldKey) else {
+                XCTFail("There is no string saved for \(oldKey)")
+                return
+            }
+
+            XCTAssertEqual(string, UserFeaturePreference.enabled.rawValue)
+            XCTAssertNil(profile.prefs.boolForKey(newKey))
+        }
+    }
+
+    func testSettingKeysToAVarietyOfSettings() {
+        verifyNilStateForKeys()
+
+        let randomSettingOptions = [true, false, nil]
+
+        keyDictionary.forEach { oldKey, newKey in
+            let oldKey = oldKey + userPrefsSuffix
+            let newKey = newKey
+
+            let randomSetting = randomSettingOptions.randomElement()
```

Comment:
Since it's random, test result isn't always the same 🤔 ?

Reply:
So, the test result is always the same because down the line, I use the random setting to test that it in fact got set to that. I could do this manually with hardcoded values, if you think that's better. I wanted it to be a more comprehensive test (ie. test various cases as we run tests), which is why I did it this way.